test/distributed/_composable/fsdp/_test_fully_shard_common.py

```diff
@@ -53,3 +55,23 @@ def patch_reduce_scatter(new_reduce_scatter_tensor: Callable):
         yield
     finally:
         dist.reduce_scatter_tensor = orig_reduce_scatter
+
+
+@contextlib.contextmanager
+def patch_unshard(new_unshard: Callable):
+    orig_unshard = FSDPParamGroup.unshard
+    FSDPParamGroup.unshard = new_unshard
```

Comment:
I do not have a better way of doing this patching context manager because of this assignment into the original method. If anyone has ideas, I can try.

Otherwise for now, we have boilerplate for every patch context.